### PR TITLE
Fixed qualifications

### PR DESCRIPTION
**Pull request by lukeelten**
@sjoerdbr 
Please implement new tests.

### CODE REVIEW

**Comment by Lehren** [comment]
One small thing that should be added:

If a request returns a 404, so no results for this specific combination of filters, the table should reset to being empty.

**Comment by Lehren** [comment]
Did it

File: src/app/components/qualification-overview-component/qualification-overview.component.ts [hunk 1]
```diff
@@ -123,12 +121,8 @@ export class QualificationOverviewComponent implements AfterContentInit, OnInit,
...
     if (this.selectedCurriculum > 0 && this.selectedLifecycleActivity > 0 && this.selectedArchitecturalLayer > 0) {
       this.backendService.getQualificationTable(this.selectedCurriculum, this.selectedArchitecturalLayer, this.selectedLifecycleActivity)
-        .subscribe(table => {
-          this.table = table;
-          return this.countTotalLearningGoals(this.table);
-        }, () => {
-          this.dataStructure = null;
-        });
+        .subscribe(table => this.countTotalLearningGoals(table),
+        () => this.dataStructure = null);
```
Comment (lukeelten) [inline src/app/components/qualification-overview-component/qualification-overview.component.ts]: I am not sure if dataStructure setting to null won't fail the table generation.

I think it is better to set it to an empty array and add furthermore an initialization to the constructor.

**Review by NNieuwenhuis — APPROVED** [review]
works perfect

**Pull request merged**